### PR TITLE
BatSet.is_singleton

### PR DESCRIPTION
**Pull request by UnixJunkie**
Sorry, the diff is "polluted" by emacs' rm_trailing_whitespaces.

### CODE REVIEW

**Comment by UnixJunkie** [comment]
related to https://github.com/ocaml-batteries-team/batteries-included/issues/1040

File: src/batteries_compattest.ml [hunk 1]
```diff
@@ -1,134 +0,0 @@
...
-  let _ = assert ([1,1;2,2;3,3;] = (sort_map (module IntSplayMap) [3; 1; 2;]))
- 
-  (* test compatibility of BatSet.S with Legacy.Set.S *)
-  let sort (type s) (module Set : Legacy.Set.S with type elt = s) l =
-      Set.elements (List.fold_right Set.add l Set.empty)
-  module IntSet = struct
-      include BatSet.Int
-  end 
-  let _ = assert ([1;2;3] = (sort (module IntSet) [3; 1; 2;]))
-end
```
Comment (gasche) [inline src/batteries_compattest.ml:134]: Why is this in the patch?

**Comment by UnixJunkie** [comment]
I will send another PR with a cleaner history.

**Comment by UnixJunkie** [comment]
superseded by https://github.com/ocaml-batteries-team/batteries-included/pull/1042

**Comment by gasche** [comment]
(Note: you can always rebase your local branch and force-push to the github remote to update a PR.)

**Pull request closed without merging**